paymaster-analysis/alchemy/alchemy-account.ts

```diff
@@ -0,0 +1,290 @@
+
+const sponsoredUserOperation: UserOperation = {
+  sender: senderAddress,
+  nonce: newNonce,
+  initCode: contractCode ? "0x" : initCode,
+  callData: encodeCallData({
+    to: senderAddress,
+    data: "0xe75235b8", // getThreshold() of the Safe. TODO: Check if this could be removed.
+    value: 0n,
+  }),
+  callGasLimit: 0n, // All Gas Values will be filled by Paymaster Response Data
+  verificationGasLimit: 0n,
+  preVerificationGas: 0n,
+  maxFeePerGas: 0n,
+  maxPriorityFeePerGas: 0n,
+  paymasterAndData: "0x",
+  signature: "0x",
+};
+
+sponsoredUserOperation.signature = await signUserOperation(
+  sponsoredUserOperation,
+  signer,
+);
+console.log("\nSigned Dummy Data for Paymaster Data Creation from Alchemy.");
+
+const gasOptions = {
+  method: "POST",
+  headers: { accept: "application/json", "content-type": "application/json" },
+  body: JSON.stringify({
+    id: 1,
+    jsonrpc: "2.0",
+    method: "alchemy_requestGasAndPaymasterAndData",
+    params: [
+      {
+        policyId: policyID,
+        entryPoint: ENTRY_POINT_ADDRESS,
+        dummySignature: sponsoredUserOperation.signature,
+        userOperation: {
+          sender: sponsoredUserOperation.sender,
+          nonce: "0x" + sponsoredUserOperation.nonce.toString(16),
+          initCode: sponsoredUserOperation.initCode,
+          callData: sponsoredUserOperation.callData,
+        },
+      },
+    ],
+  }),
+};
+
+let responseValues;
+
+if (chain == "sepolia") {
+  await fetch("https://eth-sepolia.g.alchemy.com/v2/" + apiKey, gasOptions)
+    .then((response) => response.json())
+    .then((response) => (responseValues = response))
+    .catch((err) => console.error(err));
+} else if (chain == "goerli") {
+  await fetch("https://eth-goerli.g.alchemy.com/v2/" + apiKey, gasOptions)
+    .then((response) => response.json())
+    .then((response) => (responseValues = response))
+    .catch((err) => console.error(err));
+} else {
+  throw new Error(
+    "Current code only support Sepolia and Goerli. Please make required changes if you want to use custom network.",
+  );
+}
+console.log("\nReceived Paymaster Data from Alchemy.");
+
+sponsoredUserOperation.preVerificationGas =
+  responseValues.result.preVerificationGas;
+sponsoredUserOperation.preVerificationGas =
+  responseValues.result.preVerificationGas;
+sponsoredUserOperation.callGasLimit = responseValues.result.callGasLimit;
+sponsoredUserOperation.verificationGasLimit =
+  responseValues.result.verificationGasLimit;
+sponsoredUserOperation.paymasterAndData =
+  responseValues.result.paymasterAndData;
+sponsoredUserOperation.maxFeePerGas = responseValues.result.maxFeePerGas;
+sponsoredUserOperation.maxPriorityFeePerGas =
+  responseValues.result.maxPriorityFeePerGas;
+
+sponsoredUserOperation.signature = await signUserOperation(
+  sponsoredUserOperation,
+  signer,
+);
+console.log("\nSigned Real Data including Paymaster Data Created by Alchemy.");
+
+// console.log(sponsoredUserOperation);
```

Comment:
Looks like some left-over code that should be removed.

Reply:
Done 👍🏾